src/node/routes/login.ts

```diff
@@ -12,8 +12,8 @@ export enum Cookie {
 }
 
 // RateLimiter wraps around the limiter library for logins.
-// It allows 2 logins every minute and 12 logins every hour.
-class RateLimiter {
+// It allows 2 logins every minute plus 12 logins every hour.
```

Comment:
I think that the expected behavior for a ratelimiter would match the former comment; instead of changing the comment, it may make more sense to change the behavior to match, eg.

```ts
  public try(): boolean {
    if (this.hourLimiter.getTokensRemaining() > 0) {
      return this.minuteLimiter.tryRemoveTokens(1) && this.hourLimiter.tryRemoveTokens(1)
    }
    return false;
  }
```

Reply:
Ah, okay. @code-asher and I discussed this on Slack and were a bit confused. So the comment behavior was right, but the implementation was wrong. Got it!

Reply:
IMO we should keep the original implementation. Only twelve logins an hour seems overly restrictive to me.

Reply:
We could raise it to 30 logins per hour; the way the logic I suggested works is like;
fail 1, fail 2, 60 second ratelimit, fail 3, fail 4, 60s ratelimit... repeat until n per hour (12, could increase it to 30 for now)

In the future, we should probably ratelimit by incoming IP address so that you can't deny service by just spamming the login endpoint; will create an issue about it.

Reply:
Ah I think maybe we're talking about different rate limit algorithms.

These are token buckets so if I understand correctly the current implementation should get you:

fail 1..14, minute timeout, fail 1, fail 2, minute timeout, etc

With the `&&` you could get locked out for an hour if you use up all the tokens in the bucket. Or in other words:
fail 1...14, locked out for an hour

Reply:
I got the `&&` wrong, you'd get locked out for an hour after six minutes (if you tried twice per minute).

Reply:
So where did we land on this? Do I implement with @oxy's changes, or increase to 30 logins per hour?

Reply:
I think I would prefer the original implementation since it feels a little less abusable (a service denial attack would require you to constantly log in every minute rather than just up to the hourly amount).

I think the idea behind the current implementation is to allow a steady but small stream of attempts but also support an occasional burst (the twelve).

Reply:
I think the ideal way to implement that is actually a single limiter that accrues two tokens per minute up to a maximum but it doesn't look like this API supports that, thus the two separate limiters.

Reply:
Sounds good to me! I'll go with that and then have you two double-check when ready for a second review.